File: core/src/main/java/nl/sjtek/control/core/utils/Executor.java
package nl.sjtek.control.core.utils;

import java.io.IOException;

public class Executor {

    /**
     * Execute a command.
     *
     * @param command
     * @throws IOException
     * @throws InterruptedException
     */
    public static void execute(String[] command) throws IOException, InterruptedException {
        Process process = Runtime.getRuntime().exec(command);
        process.waitFor();
        int exitValue = process.exitValue();
        if (exitValue != 0) {
            System.err.println("Exit code: " + exitValue);
        }
    }

    public static void execute(String command) throws IOException, InterruptedException {
        Process process = Runtime.getRuntime().exec(command);
        process.waitFor();
        int exitValue = process.exitValue();
        if (exitValue != 0) {
            System.err.println("Exit code: " + exitValue);
        }
    }
}

File: core/src/main/kotlin/nl/sjtek/control/core/Executor.kt
package nl.sjtek.control.core

import org.slf4j.LoggerFactory

object Executor {
    private val logger = LoggerFactory.getLogger(javaClass)

    fun execute(command: Array<String>): Int {
        try {
            val process = Runtime.getRuntime().exec(command)
            process.waitFor()
            return process.exitValue()
        } catch (e: Exception) {
            logger.error("Execution error", e)
            return 1
        }
    }
}